### PR TITLE
@W-20592464 SCAPI Schemas List and SCAPI Custom APIs Status MCP tools

### PR DESCRIPTION
**Pull request by yhsieh1**
## Summary

This PR defines the **`scapi_custom_apis_status`** MCP tool to get the status of Custom API endpoints (which are active, which failed to register). **Remote only**—no local cartridge scan or schema retrieval. Requires OAuth (sfcc.custom-apis scope). For listing/fetching SCAPI schemas (standard APIs), use **`scapi_schemas_list`**.

Test coverage script is enabled.

## Two SCAPI Tools (Scope of This PR)

The MCP exposes **two** SCAPI-related tools. This PR and doc cover only the second one:

| Tool | Purpose | When to use |
|------|---------|-------------|
| **`scapi_schemas_list`** | List or fetch SCAPI schemas (both **standard** and **custom**). Two modes: discovery (metadata) and fetch (full OpenAPI schema). Use `apiFamily: "custom"` for custom APIs. | "What SCAPI APIs are available?", "Show me shopper-products schema", "List custom API definitions", filtering by apiFamily / apiName / apiVersion. |
| **`scapi_custom_apis_status`** *(this PR)* | Get registration status of Custom API endpoints (active / not_registered). Remote only. | "What's the status of my custom API endpoints?", "Which endpoints are active?" |

### Prerequisites

- MCP server built: `pnpm --filter @salesforce/b2c-dx-mcp run build`
- MCP server running with a toolset that includes the tool (e.g. SCAPI, PWAV3, or STOREFRONTNEXT)
- **For remote tests:** Valid OAuth configured (tenantId, shortCode, sfcc.custom-apis scope), e.g. via `dw.json`, env vars, or CLI args
- MCP configured in Claude Desktop or Cursor

**MCP configuration example:**

```json
{
  "mcpServers": {
    "b2c-dx": {
      "command": "/path/to/packages/b2c-dx-mcp/bin/dev.js",
      "args": [
        "--working-directory",
        "/path/to/your-cartridge-or-storefront-project",
        "--toolsets",
        "SCAPI",
        "--allow-non-ga-tools"
      ]
    }
  }
}
```

#### 1. Basic Functionality

**Scenario:** List custom APIs with default parameters.

**Test:** Prompt "What custom APIs are in this project?" or "List custom SCAPI endpoints."

**Expected Behavior:**
- Tool returns merged remote + local endpoints (if OAuth configured)

#### 2. Status Filtering

**Scenario:** Filter endpoints by deployment status.

**Test:** Prompt "Show me only active custom APIs" or "Which custom APIs failed to deploy?"

**Expected Behavior:**
- With `status: "active"`: Returns only active endpoints (remote-sourced endpoints have `status: "active"`; local-only may still appear as `"not_registered"`)
- With `status: "not_registered"`: Returns only non-registered endpoints (includes local-only and failed remote registrations)
- Filtering applies correctly to merged results

#### 3. Schema Inclusion

**Scenario:** Include full OpenAPI schemas in response.

**Test:** Prompt "Get custom APIs with full OpenAPI schemas" or "List custom APIs and include their schema definitions."

**Expected Behavior:**
- Tool called with `includeSchemas: true`
- Each endpoint includes `schema` field with OpenAPI YAML content
- Response includes `schemasRetrieved` count
- Large schemas are capped appropriately; invalid YAML skipped without breaking response
- If WebDAV fails for remote: `webdavError` may be set; local schemas still included

#### 4. Error Handling

**Scenario:** Handle missing credentials and empty results gracefully.

**Test Cases:**
- **No OAuth:** Working directory with local cartridges → returns local endpoints only, `remoteError` set, `remoteCount: 0`
- **No local APIs:** OAuth configured, no local schemas → returns remote endpoints only, `localCount: 0`
- **No APIs at all:** No OAuth, no local schemas → empty result with `total: 0`, appropriate error message

**Expected Behavior:**
- No crashes on missing credentials or empty results
- Clear error messages via `remoteError` or `message` fields
- Tool handles all edge cases gracefully

---

- [ ] Tests pass (`pnpm test`)
- [ ] Code is formatted (`pnpm run format`)

### CODE REVIEW

File: packages/b2c-dx-mcp/src/tools/scapi/scapi-custom-apis-status.ts [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+          .string()
+          .optional()
+          .describe(
+            'Comma-separated fields to return, e.g. type,apiName,status,siteId. Valid: type, apiName, apiVersion, cartridgeName, endpointPath, httpMethod, status, siteId, securityScheme, operationId, schemaFile, implementationScript, errorReason, id.',
+          ),
+        extended: z
+          .boolean()
+          .optional()
+          .describe(
+            'If true, return all available fields (14 fields total). If false or omitted, return default columns only (6 fields: type, apiName, endpointPath, httpMethod, status, siteId). Use extended: true when you need additional details like apiVersion, cartridgeName, securityScheme, operationId, schemaFile, implementationScript, errorReason, or id. Note: If you only need specific fields, use columns parameter instead for more control.',
```
Comment (patricksullivansf) [inline packages/b2c-dx-mcp/src/tools/scapi/scapi-custom-apis-status.ts]: columns and extended seem to do largely the same thing: decide what columns are returned. We could make the LLM have an easier chore by giving them one option instead of two.

File: packages/b2c-dx-mcp/src/tools/scapi/scapi-custom-apis-status.ts [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+ * Creates the scapi_custom_apis_status tool.
+ *
+ * Mirrors CLI: b2c scapi custom status. All flags supported; agent chooses what to use.
+ * See: https://salesforcecommercecloud.github.io/b2c-developer-tooling/cli/custom-apis.html#b2c-scapi-custom-status
+ */
+export function createScapiCustomApisStatusTool(services: Services): McpTool {
+  return createToolAdapter<CustomListInput, CustomListOutput>(
+    {
+      name: 'scapi_custom_apis_status',
+      description: `List Custom SCAPI API endpoints and their registration status (active vs not_registered). Returns individual HTTP endpoints (e.g., GET /hello, POST /items/{id}) with deployment status, one row per endpoint per site. Use this for developer-defined custom APIs only.
```
Comment (patricksullivansf) [inline packages/b2c-dx-mcp/src/tools/scapi/scapi-custom-apis-status.ts]: although i like the extended description, we should keep in mind that tool descriptions eat context tokens for EVERY request, even when not used.  Across dozens if tools this can get unwieldy.We might want to ask if the LLM can compress this a bit.

File: packages/b2c-dx-mcp/src/tools/scapi/scapi-custom-apis-status.ts [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+import {getApiErrorMessage} from '@salesforce/b2c-tooling-sdk/clients';
+import {getApiType} from '@salesforce/b2c-tooling-sdk/operations/scapi';
+
+type CustomApiEndpoint = CustomApisComponents['schemas']['CustomApiEndpoint'];
+
+/** Endpoint with optional display field (type) added by the tool. */
+type EndpointWithMeta = CustomApiEndpoint & {type?: string};
+
+const DEFAULT_COLUMNS = ['type', 'apiName', 'endpointPath', 'httpMethod', 'status', 'siteId'] as const;
+const ALL_COLUMN_KEYS = [
```
Comment (patricksullivansf) [inline packages/b2c-dx-mcp/src/tools/scapi/scapi-custom-apis-status.ts]: weird to me these aren't defined at the SDK level.

File: packages/b2c-dx-mcp/src/tools/scapi/scapi-schemas-list.ts [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+  client: ReturnType<Services['getScapiSchemasClient']>;
+  organizationId: string;
+  args: SchemasListInput;
+  shortCode?: string;
+}): Promise<SchemaGetOutput> {
+  const {client, organizationId, args, shortCode} = params;
+  const {apiFamily, apiName, apiVersion, expandAll, status} = args;
+
+  // Warn if status filter was provided (it's ignored in fetch mode)
+  const warning = status
```
Comment (patricksullivansf) [inline packages/b2c-dx-mcp/src/tools/scapi/scapi-schemas-list.ts]: for my info, what does the LLM typically do when it gets this warning mixed into the response?

File: packages/b2c-dx-mcp/src/tools/scapi/scapi-schemas-list.ts [hunk 5]
```diff
@@ -0,0 +1,404 @@
...
+  - "Show me the shopper-products v1 OpenAPI schema" → apiFamily: "product", apiName: "shopper-products", apiVersion: "v1", includeSchemas: true.
+- Custom APIs:
+  - "What custom API definitions are available?" → list with apiFamily: "custom".
+  - "Show me the loyalty-points custom API schema" → apiFamily: "custom", apiName: "loyalty-points", apiVersion: "v1", includeSchemas: true.
+- General:
+  - "List all current schemas" → list with status: "current".
+
+**Requirements:** Instance must have shortCode, tenantId, and OAuth with sfcc.scapi-schemas scope.`,
+      toolsets: ['PWAV3', 'SCAPI', 'STOREFRONTNEXT'],
+      isGA: true,
```
Comment (patricksullivansf) [inline packages/b2c-dx-mcp/src/tools/scapi/scapi-schemas-list.ts]: do we really want to go straight to GA?

File: packages/b2c-dx-mcp/src/tools/scapi/scapi-schemas-list.ts [hunk 6]
```diff
@@ -0,0 +1,404 @@
...
+          .describe('API name (e.g., "shopper-baskets", "shopper-products"). Use to filter or identify schemas.'),
+        apiVersion: z
+          .string()
+          .optional()
+          .describe('API version (e.g., "v1", "v2"). Use to filter or identify schemas.'),
+        status: z
+          .enum(['current', 'deprecated'])
+          .optional()
+          .describe(
+            'Filter by schema status. "current" returns only active schemas, "deprecated" returns deprecated ones. Only works in discovery mode.',
```
Comment (patricksullivansf) [inline packages/b2c-dx-mcp/src/tools/scapi/scapi-schemas-list.ts]: which is better, default, etc? or does the LLM do a good job of guessing?

File: packages/b2c-dx-mcp/src/services.ts [hunk 7]
```diff
@@ -170,20 +217,102 @@ export class Services {
...
+   *
+   * @throws Error if tenantId is not configured
+   * @returns Organization ID with f_ecom_ prefix
+   */
+  public getOrganizationId(): string {
+    const {tenantId} = this.resolvedConfig.values;
+
+    if (!tenantId) {
+      throw new Error(
+        'Tenant ID required. Provide --tenant-id, set SFCC_TENANT_ID, or configure tenant-id in dw.json.',
```
Comment (patricksullivansf) [inline packages/b2c-dx-mcp/src/services.ts]: love the descriptive error with possible corrections

File: packages/b2c-dx-mcp/test/tools/scapi/scapi-custom-apis-status.test.ts [hunk 8]
```diff
@@ -0,0 +1,424 @@
...
+      expect(tool.inputSchema).to.have.property('status');
+      expect(tool.inputSchema).to.have.property('groupBy');
+      expect(tool.inputSchema).to.have.property('columns');
+      expect(tool.inputSchema).to.have.property('extended');
+    });
+  });
+
+  describe('handler', () => {
+    it('should return endpoints and total when API returns data', async () => {
+      mockGet.resolves({
```
Comment (patricksullivansf) [inline packages/b2c-dx-mcp/test/tools/scapi/scapi-custom-apis-status.test.ts]: I would rather MCP tests didn't include the SDK, but there are arguments either way.

pro to mocking API response - guarantees integration from API response to MCP output is verified.

con makes the MCP tests brittle and sensitive to SDK implementation, instead of relying on SDK to enforce it's contract with consumers to adapt and hide API changes.

**Review by patricksullivansf — COMMENTED** [review]
Aside:
  
> 5. Tool Selection
> Scenario: Ensure correct tool is used for schemas vs endpoint status.

Rather than part of a tool implementation PR, I think tool selection testing is more reliably to be done as a separate task, ideally with dozens of tools enabled.

File: packages/b2c-dx-mcp/test/tools/scapi/scapi-custom-apis-status.test.ts (hunk 8 above)
Comment (clavery) [reply packages/b2c-dx-mcp/test/tools/scapi/scapi-custom-apis-status.test.ts]: I agree. If you can mock the SDK the better. We have good test coverage there. Coverage here should be on MCP linnes

**Review by patricksullivansf — APPROVED** [review]

File: .changeset/scapi-discovery-tools.md
[the remark this answers is not in the document]
Comment (yhsieh1) [reply .changeset/scapi-discovery-tools.md]: @patricksullivansf changeset is updated. Please check.

File: packages/b2c-tooling-sdk/src/operations/scapi/index.ts
[the remark this answers is not in the document]
Comment (yhsieh1) [reply packages/b2c-tooling-sdk/src/operations/scapi/index.ts]: Thanks. I will move it there.

**Review by patricksullivansf — APPROVED** [review]

**Review by clavery — DISMISSED** [review]
